fastlane/Fastfile

```diff
@@ -596,6 +596,15 @@ platform :ios do
       )
     end
 
+    # We're about to use `add_development_certificates_to_provisioning_profiles` and `add_all_devices_to_provisioning_profiles`.
+    # These actions use Developer Portal APIs that don't yet support authentication via API key (-.-').
+    # Let's preemptively ask for and set the email here to avoid being asked twice for it if not set.
+    asc_user_email_key = 'FASTLANE_USER'
+    unless ENV.key?(asc_user_email_key)
```

Comment:
One drawback I see from doing the test like this is if the user defined their fastlane user in any other place:
 - Like if a username was defined in a `Matchfile` or `Appfile`
 - Or if the user used fastlane's [`CredentialManager` — whose doc can be found here](https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/credentials_manager/README.md) — to store the username

Which means that if I defined my username in one of those alternate ways, your lane will prompt me (`UI.input`) for my username anyway even if it's already defined by some other mean that `match` would have found and used

---

I'm sure fastlane has some methods or classes we can call to check if a username is defined (and would be used by default by `match`), wherever it might be defined (ENV, file, …), and using that would probably be a more resilient solution.

I just didn't take the time to dig deep into fastlane's codebase to find what class or method would achieve that, but maybe it could be worth taking a quick look to see if such a thing exists? [EDIT: Maybe calling [`CredentialManager.new.user`](https://github.com/fastlane/fastlane/blob/b8045bb6ab964bd334a2c87a297ed3a7f192eb4f/credentials_manager/lib/credentials_manager/account_manager.rb#L30-L35) would do the trick? 🤔 ]

(PS: if, after giving the above a try, you can't make it work in a reasonable time, I think it would still be acceptable to assume that most of us use `export FASTLANE_USER=…` in their `~/.zshrc` and that the test here would be good enough — instead of spending too much time in finding a more resilient solution. But it's maybe still worth a try to see if some quick alternative solutions would work first, before bailing before sinking into a rabbit hole?)